### PR TITLE
Refactored configuration implementation

### PR DESCRIPTION
**Pull request by Subomi**
### Problems

- Enhanced readability for the entry point of the system. Currently, It’s pretty hard to follow through with how the system is set up from the `cmd` package. The various commands don’t have a consistent way they build `Configuration`. Here’s how the `server` works today;
    - `convoy.json` → `environment variables` → `CLI Flags` → `Server CLI Flags` → `Server Defaults`
- Simpler to the above, provide a consistent approach for `subcommands` to add override configuration rather than spin up ad-hoc mechanisms.
### Solution

1. Build `Configuration` with a more straightforward flow and API.
    - Flow
        
        `defaults` → `convoy.json` → `environment variable` → `CLI flags`
        
    - API
        
        ```go
        // Path to convoy.json
        cfgPath := "path/to/convoy.json"
        
        // Load default configuration
        err := config.LoadConfig(cfgPath)
        
        // Get configuration
        c, err := config.Get()
        
        // Override configuration
        err := config.Override(newConfig)
        ```
        
2. Create `convoy config` to output a JSON dump of the current shape of `Configuration` for debugging sake.

### CODE REVIEW

**Comment by kwonsd-hwik** [comment]
34234

**Review by Dotunj — APPROVED** [review]

**Review by jirevwe — APPROVED** [review]
